webui/react/src/pages/ExperimentDetails/ExperimentActions.tsx

```diff
@@ -29,91 +29,68 @@ interface Props {
   onSettled: () => void; // A callback to trigger after an action is done.
 }
 
-type ButtonLoadingStates = Record<Action, boolean>;
-
 const ExperimentActions: React.FC<Props> = ({ experiment, onClick, onSettled }: Props) => {
-  const [ btnLoadingStates, setBtnLoadingStates ] = useState<ButtonLoadingStates>({
-    Activate: false,
-    Archive: false,
-    Cancel: false,
-    Fork: false,
-    Kill: false,
-    Pause: false,
-    Tensorboard: false,
-    Unarchive: false,
-  });
+  const [ isRunningActivate, setIsRunningActivate ] = useState<boolean>(false);
```

Comment:
question: any downsides you saw with the way loadingStates were being tracked before?

Reply:
I think it's a good practice to flatten data structure, because otherwise changing any of the multiple states will require to compose a new object. In that case, practically, the 2 `useEffect` would need to be dependent on that object, increasing the probability of errors and the complexity of code.